src/main/java/co/elastic/logstash/filters/elasticintegration/EventProcessorBuilder.java

```diff
@@ -307,37 +313,42 @@ public synchronized EventProcessor build(final PluginContext pluginContext) {
         Map<String, ScriptEngine> engines = new HashMap<>();
-        engines.put(PainlessScriptEngine.NAME, new PainlessScriptEngine(settings, scriptContexts));
+        engines.put(PainlessScriptEngine.NAME, getPainlessScriptEngine(settings));
         engines.put(MustacheScriptEngine.NAME, new MustacheScriptEngine());
         return new ScriptService(settings, engines, ScriptModule.CORE_CONTEXTS, threadPool::absoluteTimeInMillis);
     }
 
     /**
-     * @implNote handles breaking changes introduced in Elasticsearch 8.14 series; once 8.14 is
-     *           released and all builds of this plugin depend on Elasticsearch 8.14+, this can
-     *           be simplified to call {@code PainlessPlugin.baseWhiteList()} directly.
-     * @return the PainlessPlugin's default base whitelists
+     * @param settings the Elasticsearch settings object
+     * @return a {@link ScriptEngine} for painless scripts for use in {@link IngestScript} and
+     *         {@link IngestConditionalScript} contexts, including all available {@link PainlessExtension}s.
+     * @throws IOException when the underlying script engine cannot be created
      */
-    @SuppressWarnings({"JavaReflectionMemberAccess", "unchecked"})
-    static List<Whitelist> getPainlessBaseWhiteList() {
-        Class<PainlessPlugin> cls = PainlessPlugin.class;
-        try {
-            try {
-                // In 8.14+: PainlessPlugin.baseWhiteList()
-                return (List<Whitelist>) cls.getMethod("baseWhiteList").invoke(null);
-            } catch (NoSuchMethodException e) {
-                // in 8.x->8.13.x: PainlessPlugin.BASE_WHITELISTS
-                return (List<Whitelist>) cls.getField("BASE_WHITELISTS").get(null);
-            }
-        } catch (java.lang.reflect.InvocationTargetException | IllegalAccessException | NoSuchFieldException e) {
-            throw new RuntimeException("Unsupported PainlessPlugin does not provide access to its base whitelists", e);
+    private static ScriptEngine getPainlessScriptEngine(final Settings settings) throws IOException {
+        try (final PainlessPlugin painlessPlugin = new PainlessPlugin()) {
+
+            painlessPlugin.loadExtensions(new ExtensiblePlugin.ExtensionLoader() {
+                @Override
+                @SuppressWarnings("unchecked")
+                public <T> List<T> loadExtensions(Class<T> extensionPointType) {
+                    if (extensionPointType.isAssignableFrom(PainlessExtension.class)) {
+                        final List<PainlessExtension> extensions = new ArrayList<>();
+
+                        extensions.add(new ConstantKeywordPainlessExtension()); // module: constant-keyword
+                        extensions.add(new ProcessorsWhitelistExtension());     // module: ingest-common
+                        extensions.add(new SpatialPainlessExtension());         // module: spatial
+                        extensions.add(new WildcardPainlessExtension());        // module: wildcard
```

Comment:
@mashhurs you brought up on a separate channel that we should audit to see what other painless extensions need to be registered, so I went through all of the implementations of `PainlessExtension` in ES source to find those that worked with either the `IngestScript.CONTEXT` or `IngestConditionalScript.CONTEXT`:

| `PainlessExtension` | Script Context | Source |
| -------------------------------- | ---------------- | ------------------- |
| `AggregationsPainlessExtension`    | ❌context-miss  | --                  |
| `AnalysisPainlessExtension`        | ❌context-miss  | --                  |
| `ProcessorsWhitelistExtension`     | ✅context-match | ✅ingest-common     |
| `RuntimeFieldsPainlessExtension`   | ❌context-miss  | --                  |
| `Murmur3PainlessExtension`         | ✅context-match | _❌not included_    |
| `EqlPainlessExtension`             | ❌context-miss  | --                  |
| `ConstantKeywordPainlessExtension` | ✅context-match | ✅constant-keyword  |
| `DocValuesWhitelistExtension`      | ❌context-miss  | --                  |
| `VersionFieldDocValuesExtension`   | ❌context-miss  | --                  |
| `MachineLearningPainlessExtension` | ❌context-miss  | --                  |
| `SpatialPainlessExtension`         | ✅context-match | ✅spatial           |
| `SqlPainlessExtension`             | ❌context-miss  | --                  |
| `WatcherPainlessExtension`         | ❌context-miss  | --                  |
| `WildcardPainlessExtension`        | ✅context-match | ✅wildcard          |

The murmur3 painless plugin _would_ work, but from what I can tell it isn't included in the ES default distribution.